meilisearch-http/src/option.rs

```diff
@@ -4,127 +4,127 @@ use std::path::PathBuf;
 use std::sync::Arc;
 
 use byte_unit::Byte;
+use clap::Parser;
 use meilisearch_lib::options::IndexerOpts;
 use rustls::internal::pemfile::{certs, pkcs8_private_keys, rsa_private_keys};
 use rustls::{
     AllowAnyAnonymousOrAuthenticatedClient, AllowAnyAuthenticatedClient, NoClientAuth,
     RootCertStore,
 };
-use structopt::StructOpt;
 
 const POSSIBLE_ENV: [&str; 2] = ["development", "production"];
 
-#[derive(Debug, Clone, StructOpt)]
+#[derive(Debug, Clone, Parser)]
 pub struct Opt {
     /// The destination where the database must be created.
-    #[structopt(long, env = "MEILI_DB_PATH", default_value = "./data.ms")]
+    #[clap(long, env = "MEILI_DB_PATH", default_value = "./data.ms")]
     pub db_path: PathBuf,
 
     /// The address on which the http server will listen.
-    #[structopt(long, env = "MEILI_HTTP_ADDR", default_value = "127.0.0.1:7700")]
+    #[clap(long, env = "MEILI_HTTP_ADDR", default_value = "127.0.0.1:7700")]
     pub http_addr: String,
 
     /// The master key allowing you to do everything on the server.
-    #[structopt(long, env = "MEILI_MASTER_KEY")]
+    #[clap(long, env = "MEILI_MASTER_KEY")]
     pub master_key: Option<String>,
 
     /// This environment variable must be set to `production` if you are running in production.
     /// If the server is running in development mode more logs will be displayed,
     /// and the master key can be avoided which implies that there is no security on the updates routes.
     /// This is useful to debug when integrating the engine with another service.
-    #[structopt(long, env = "MEILI_ENV", default_value = "development", possible_values = &POSSIBLE_ENV)]
+    #[clap(long, env = "MEILI_ENV", default_value = "development", possible_values = &POSSIBLE_ENV)]
     pub env: String,
 
     /// Do not send analytics to Meili.
     #[cfg(all(not(debug_assertions), feature = "analytics"))]
-    #[structopt(long, env = "MEILI_NO_ANALYTICS")]
-    pub no_analytics: Option<Option<bool>>,
+    #[clap(long, env = "MEILI_NO_ANALYTICS")]
+    pub no_analytics: bool,
```

Comment:
I am at the regret to say that you must keep it like before, use an `Option<Option<bool>>` to keep the correct behavior: being able to specify `true` or `false` or nothing to enable the analitycs.

Reply:
The goal is not to have to specify true or false, i.e is no analytics is not passed, then analytics are enabled, and to disable them you need to pass the flag

Reply:
@nicolasvienot I would like to have your opinion on this 🤓

> All flag arguments now take NO argument, i.e:
> meilisearch --schedule-snapshot true becomes meilisearch --schedule-snapshot

What do you think of this? Do you see a limit in your use of the CLI, I'm thinking mostly of the manipulation of containers dynamically within kubernetes.

Reply:
Yup, by using an `Option<Option<bool>>` you do both: specify `true` or `false` and also not specify anything. It is the behavior explained in the documentation and the behavior that @nicolasvienot wanted for the SaaS. Waiting for Guigui about what we should do, keep or change.

Reply:
Hey @gmourier,

I don't think this is an issue for the manipulation of containers dynamically within Kubernetes or MeiliSearch Cloud.
We should be able to add the flag conditionally, even if this was a bit easier with `true` / `false`.

As long as this seems better for the core team and it's the same behavior everywhere, I see no objection. 😃 

@Kerollmops I made the issue about the `MEILI_NO_ANALYTICS` because I was following the doc and it seemed wrong, but we actually handle both for now because we still support < `0.25.0` versions.

Reply:
@nicolasvienot, there is a misanderstanding. For the environment variable, the behaviour is the following:
> A false literal is n, no, f, false, off or 0. An absent environment variable will also be considered as false. Anything else will considered as true.
that means that a set empty env variable is considered a true as well

for the cli flag, the flag commands do not accept an argument anymore.

If you plan on only using the env variables, this should not change anything for you.

Reply:
@MarinPostma Thanks for the clarification. I just checked and as you said, since we are using env variables for now, this will still work fine. :seal:

Reply:
Thanks, @nicolasvienot, and @MarinPostma!

@curquiza Do you think we can plan it for v0.26.0?